docs/internal.md

```diff
@@ -0,0 +1,47 @@
+# Internal
+
+This documentation describes the inner design concepts of gallia.
+
+The following diagram shows the internal modules of gallia and how they are connected:
+![tooling-architecture.png](tooling-architecture.png)
+
+## Scanner Lifecycle
+
+This section describes the livecycle of a scanner.
+Meaning, which methods are called in which order when executing a scanner.
+
+1. {meth}`gallia.cli.main`
+2. {meth}`gallia.udscan.core.GalliaBase.run` 
+3. {meth}`gallia.udscan.core.GalliaBase._run`
+4. {meth}`gallia.udscan.core.Scanner.setup`
+5. {meth}`gallia.udscan.core.GalliaBase.main` 
+6. {meth}`gallia.udscan.core.Scanner.teardown`
+
+## Argument Parsing
+
+In gallia, there is a central argument parser ({attr}`gallia.udscan.core.GalliaBase.parser`), 
+where all scanner are added as sub-commands.
+Scanner can add their specific options by implementing the {meth}`gallia.udscan.core.GalliaBase.add_parser` method.
+With this approach, we can generate a usage, which lists all installed scanners.
+To see the scanner specific usage, the user can query the help of the specific sub-command.
+
+## Plugin Interface
+
+There are different plugin interfaces to extend gallia.
+All of them use the python `entry point` concept:
+
+1. Scanner API (`gallia_scanner`)
```

Comment:
We should rename this to `gallia_scanners`, since we use plural everywhere.

Reply:
When we do this, remember to rename the entry point in `pyproject.toml` and `udscan.core` as well.